File: sortega/prime-factors/src/main/java/com/sortega/primefactors/PrimeFactors.java
package com.sortega.primefactors;

import java.util.LinkedList;
import java.util.List;

public class PrimeFactors {

    public static List<Integer> generate(int number) {
        List<Integer> factors = new LinkedList<Integer>();

        int candidate = 2;
        while (candidate <= number) {
            if (divisibleBy(number, candidate)) {
                factors.add(candidate);
                number /= candidate;
            } else {
                candidate++;
            }
        }
        
        return factors;
    }

    private static boolean divisibleBy(int number, int divisor) {
        return number % divisor == 0;
    }
}

File: pasku/PHP/src/PrimeFactors.php
<?php

class PrimeFactors
{
    public static function generate($number)
    {
        $factors = array();
        for ($divisor = 2; $number > 1; $divisor++) {
            for (; $number % $divisor === 0; $number /= $divisor) {
                $factors[] = $divisor;
            }
        }
        return $factors;
    }
}
